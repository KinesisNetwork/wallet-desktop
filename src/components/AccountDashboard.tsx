import * as React from 'react'

import { Transactions } from '@containers/Transactions'
import { Transfer } from '@containers/Transfer'
import { WalletInfo } from '@containers/WalletInfo'

export const AccountDashboard: React.SFC = () => (
  <React.Fragment>
    <div className='columns is-constant-height'>
      <div className='column is-half'>
        <WalletInfo />
      </div>
      <div className='column is-half'>
        <Transfer />
      </div>
    </div>
    <Transactions />
  </React.Fragment>
)

AccountDashboard.displayName = 'AccountDashboard'
