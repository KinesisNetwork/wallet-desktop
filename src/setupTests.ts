import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
