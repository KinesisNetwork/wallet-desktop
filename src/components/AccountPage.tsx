import * as React from 'react'

import { AccountDashboard } from '@components/AccountDashboard'
import { AccountPanel } from '@containers/AccountPanel'
import { Onboarding } from '@containers/Onboarding'
import { RootState } from '@store'
import { connect } from 'react-redux'
import { Route, RouteComponentProps, Switch, withRouter } from 'react-router'

const mapStateToProps = (state: RootState) => ({
  hasOnBoarded: state.settings.onBoarding,
})

type Props = RouteComponentProps<any> & ReturnType<typeof mapStateToProps>

const AccountPagePresentation: React.SFC<Props> = ({ match, hasOnBoarded }) => (
  <React.Fragment>
    {!hasOnBoarded && <Onboarding />}
    <AccountPanel />
    <Switch>
      <Route exact={true} path={`${match.path}`} component={AccountDashboard} />
    </Switch>
  </React.Fragment>
)

const AccountPage = withRouter(connect(mapStateToProps)(AccountPagePresentation))

export { AccountPage }
