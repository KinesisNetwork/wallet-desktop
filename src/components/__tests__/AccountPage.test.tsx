import { shallow } from 'enzyme';
import * as React from 'react'

import { AccountPage } from '@components/AccountPage'
import { AccountPage as AccountPageEnum } from '@types'

import { AccountDashboard } from '@components/AccountDashboard';
import '../../setupTests'

describe('AccountPage', () => {
  it('should render properly', () => {
    const wrapperWhenAccountDashboard = shallow(<AccountPage accountPage={AccountPageEnum.dashboard} />)
    const wrapperWhenSign = shallow(<AccountPage accountPage={AccountPageEnum.sign} />)

    expect(wrapperWhenAccountDashboard).toMatchSnapshot()
    expect(wrapperWhenSign).toMatchSnapshot()
  })

  it('renders components if accountPage property is given', () => {
    const wrapperWhenAccountDashboard = shallow(<AccountPage accountPage={AccountPageEnum.dashboard} />)
    const wrapperWhenSign = shallow(<AccountPage accountPage={AccountPageEnum.sign} />)

    expect(wrapperWhenAccountDashboard.children()).toHaveLength(2)
    expect(wrapperWhenAccountDashboard.find(AccountDashboard)).toHaveLength(1)
    expect(wrapperWhenSign.find('Connect(Sign)')).toHaveLength(1)
  })
})
