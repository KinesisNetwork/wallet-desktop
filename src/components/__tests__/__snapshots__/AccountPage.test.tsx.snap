// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountPage should render properly 1`] = `
<div
  className="vertical-spaced"
>
  <div>
    <div
      className="level"
    >
      <div
        className="level-left"
      >
        <div
          className="level-item"
        >
          <Connect(AccountPageSelector) />
        </div>
      </div>
      <div
        className="level-right"
      >
        <div
          className="level-item"
        >
          <Connect(Password) />
        </div>
      </div>
    </div>
  </div>
  <AccountDashboard />
</div>
`;

exports[`AccountPage should render properly 2`] = `
<div
  className="vertical-spaced"
>
  <div>
    <div
      className="level"
    >
      <div
        className="level-left"
      >
        <div
          className="level-item"
        >
          <Connect(AccountPageSelector) />
        </div>
      </div>
      <div
        className="level-right"
      >
        <div
          className="level-item"
        >
          <Connect(Password) />
        </div>
      </div>
    </div>
  </div>
  <Connect(Sign) />
</div>
`;
